crates/subspace-gateway/src/piece_getter.rs

```diff
@@ -68,6 +70,38 @@ where
             .get_piece_from_archival_storage(piece_index, MAX_RANDOM_WALK_ROUNDS)
             .await)
```

Comment:
@nazar-pc is this (and the similar code below) still needed after your PR #3259?

Reply:
Yes, random walking is a cold storage fallback fr cases when a piece is not cached, in that case it is wandering around the network hoping to stumble upon someone storing a piece in their plot rather than cache